src/nnet3/nnet-chain-training.cc

```diff
@@ -68,23 +68,43 @@ void NnetChainTrainer::Train(const NnetChainExample &chain_eg) {
                              &request);
   const NnetComputation *computation = compiler_.Compile(request);
 
-  NnetComputer computer(nnet_config.compute_config, *computation,
+  if (nnet_config.adversarial_training_scale > 0.0 && num_minibatches_processed_
+      % nnet_config.adversarial_training_interval == 0) {
+    // adversarial training is incompatible with momentum > 0
+    KALDI_ASSERT(nnet_config.momentum == 0.0);
+    FreezeNaturalGradient(true, delta_nnet_);
+    bool is_adversarial_step = true;
```

Comment:
@danpovey For reversing the two steps, the only changes I made is setting is_adversarial_step=false here and is_adversarial_step=true at line 81.